### PR TITLE
Python 3.10 support

### PR DESCRIPTION
**Pull request by jgberry**
Adds support for structural pattern matching introduced in Python 3.10.

### CODE REVIEW

**Comment by bwhmather** [comment]
Looks good to me.  I think the only thing missing is to add 3.10 to the list of versions exercised by github actions.  If you ever happen to be in London, or thereabouts, then I think I now also owe you a pint.

**Comment by jgberry** [comment]
Ok, I'll update the actions. What do you think about upgrading all the actions to run on Python 3.10? I'd like to avoid disabling linting for Python 3.10 exclusives as I had to do in [01ad66d](https://github.com/bwhmather/ssort/pull/45/commits/01ad66d9a738247d870b59fb08709065f4b2f835).

Also, I'll have to take you up on that pint if I'm ever in the area. Although, I'm not too much of a drinker myself and I've heard you all across the pond are professionals :smile:

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -100,10 +100,10 @@ jobs:
     runs-on: ubuntu-18.04
     steps:
     - uses: actions/checkout@v2
-    - name: Set up Python 3.9
+    - name: Set up Python 3.10
       uses: actions/setup-python@v2
       with:
-        python-version: 3.9
+        python-version: 3.10
```
Comment (bwhmather) [inline .github/workflows/ci.yaml:106]: "Version 3.1 not found".  I think this needs to be a string

**Comment by bwhmather** [comment]
Coffee also good.  Running all actions using 3.10 probably makes sense.  `mypy` is usually also quite slow with support for new versions so you might need to bump that dependency.

**Comment by bwhmather** [comment]
Ignore that last bit.  Apparently I never actually added mypy to the CI config.  I was a bit dubious about adding annotations to `ssort` because it made the project feel a bit too much like work.  Should probably revisit now that there are people actually using it.

**Comment by jgberry** [comment]
I think adding mypy would be a big help. I've been trying to run it locally as much as possible to catch any issues; although, I may need to clean up some of the type annotations I added if we use it as a CI check. I'd be happy to work on that next.

**Comment by bwhmather** [comment]
That would be great.  There is also https://github.com/bwhmather/ssort/issues/23, which I suspect is biting lots of people who use venv.

**Comment by bwhmather** [comment]
(if you are wondering why I never finished that `exclude` branch, the black `files` module doesn't respect `.gitignore` files outside the project root.  There's a fair bit of work to do to make it match git, and I'm not sure whether it makes sense to do it without a plan to push the improvements back to `black`)

**Pull request merged**